### PR TITLE
feat: add notification state to the scan document interface

### PR DESCRIPTION
**Pull request by haonliu**
Adding notification state to the scan document interface

### CODE REVIEW

File: packages/storage-documents/src/on-demand-page-scan-result.ts [hunk 1]
```diff
@@ -36,6 +37,7 @@ export interface OnDemandPageScanResult extends StorageDocument {
     priority: number;
     itemType: ItemType.onDemandPageScanRunResult;
     batchRequestId: string;
+    notificationState?: NotificationState;
```
Comment (devaradhanm) [inline packages/storage-documents/src/on-demand-page-scan-result.ts]: we should have an object that contains things like notification status, error message, error code etc

Comment (lamaks) [reply packages/storage-documents/src/on-demand-page-scan-result.ts]: > notification status, error message, error code etc
are not candidates for the single object, why we need that object? Example?

Comment (devaradhanm) [reply packages/storage-documents/src/on-demand-page-scan-result.ts]: pageScanResult {
run: {...},
notification: {url: string, status: enum, error: {code, message}}
}

File: packages/storage-documents/src/on-demand-page-scan-result.ts [hunk 2]
```diff
@@ -36,6 +37,18 @@ export interface OnDemandPageScanResult extends StorageDocument {
...
+}
+
+export interface ScanCompletedNotification {
+    notificationUrl: string;
+    state?: NotificationState;
+    error?: NotificationError;
+}
+
+export interface NotificationError {
+    errorType: string;
```
Comment (devaradhanm) [inline packages/storage-documents/src/on-demand-page-scan-result.ts]: these should be in one of the known strings. (like ScanError). If not sure what all notification error types we will need, then just add one
NotificationErrorTypes= "HttpErrorCode"

**Pull request merged**